internal/server/iigo.go

```diff
@@ -35,23 +35,46 @@ func (s *SOMASServer) runIIGOTax() error {
 	clientMap := getNonDeadClients(s.gameState.ClientInfos, s.clientMap)
 	for clientID, v := range clientMap {
 		tax := v.GetTaxContribution()
-		err := s.takeResources(clientID, tax, "tax")
-		if err == nil {
+		sanction := v.GetSanctionPayment()
+		clientTaxErr := s.takeResources(clientID, tax, "tax")
```

Comment:
rewrite to handle `err != nil` branch first

Reply:
Sure, have fixed

Reply:
Should be directly in the line below

Reply:
Sure, changes pushed